internal/cluster/node.go

```diff
@@ -91,15 +102,20 @@ func (n *NodeInfo) populateNodeIpList() error {
 }
 
 func GetNodeIP(client client.Client) (string, error) {
-	nodeInfo := &NodeInfo{
-		Client: client,
-	}
-	nodeIPs, err := nodeInfo.getNodeExternalIpList(client)
-	if err != nil {
+	nodeInfo.Client = client
+	nodeIPs, err := nodeInfo.getNodeExternalIpList()
+	if err != nil || len(nodeIPs) == 0 {
```

Comment:
it is better to pass `client` as argument as in the previous code. Why is this changed?

Reply:
nodeInfo struct already has a client as a field